### PR TITLE
Wave 2: Local seam stabilization

### LINKED ISSUE #73: [Stabilization] Decouple curated visibility plots from the generic msexplore scatter engine
## Problem
The public `casa-ms` library plotting builder currently routes curated raw-visibility plots directly through the giant generic scatter engine in `msexplore.rs`. That keeps an exported library seam coupled to a very large internal module and makes raw-visibility payload regressions harder to isolate.

Hotspots:
- `crates/casa-ms/src/plot.rs`
- `crates/casa-ms/src/msexplore.rs`

## Scope
Decouple the curated raw-visibility lowering path for:
- `AmplitudeVsTime`
- `PhaseVsTime`
- `AmplitudeVsUvDistance`

into a dedicated internal library module owned by `casa-ms`, with `plot.rs` depending on that seam instead of on the giant generic scatter engine.

## Acceptance Criteria
- [x] The curated raw-visibility lowering path no longer depends directly on the oversized `msexplore.rs` implementation.
- [x] Payload structs, labels, grouping, and rendered/exported output remain unchanged for covered cases.
- [x] A fixture-backed regression test builds all three curated payloads through the library API and asserts stable payload shape, labels, and series counts.
- [x] `cargo test -p casa-ms --lib` passes.
- [x] `cargo test -p casa-ms --test ms_library_coverage` passes.
- [x] `cargo clippy -p casa-ms --all-targets -- -D warnings` passes.
- [x] `cargo fmt --all -- --check` passes.

## Non-goals
- No CLI work
- No `msexplore/cli.rs` changes
- No new presets
- No averaging or iteration redesign
- No public API renames

## Stop conditions
Stop before implementation if the change would alter payload semantics, require protocol/CLI changes, or widen scope into general `msexplore` restructuring beyond the curated raw-visibility seam.

## Closeout

Implemented: added `crates/casa-ms/src/plot_visibility.rs` as the internal curated raw-visibility lowering seam, switched `plot.rs` to use it, and removed the old listobs compatibility lowering from `msexplore.rs`.
Changed from plan: none.
Deferred / follow-ups: none.
Verification run: `cargo test -p casa-ms --lib`, `cargo test -p casa-ms --test ms_library_coverage`, `cargo clippy -p casa-ms --all-targets -- -D warnings`, `cargo fmt --all -- --check`, `just quick`, `just verify`.
PR: #149.
Branch / worktree cleanup: deferred until merge via `wdad-pr-merge`.

### LINKED ISSUE #75: [Stabilization] Extract listobs selector parsing and TAQL building from summary/rendering
## Problem
`listobs.rs` mixes selector parsing and TAQL construction with summary building and text rendering inside one large library module. That makes selector failure modes harder to test directly and couples parsing concerns to presentation concerns.

Hotspots:
- `crates/casa-ms/src/listobs.rs`

## Scope
Extract selector parsing and TAQL construction into a focused internal submodule while preserving `ListObsSummary` and `ListObsUvCoverage` behavior.

## Acceptance Criteria
- [x] Selector parsing and TAQL construction no longer live inline with summary/rendering logic.
- [x] Current happy-path selection behavior is preserved.
- [x] Direct adversarial tests cover empty timerange, invalid date/time fragments, mixed uvrange units, unmatched field globs, unmatched intents, and empty/invalid correlation selectors.
- [x] `cargo test -p casa-ms --lib` passes.
- [x] `cargo test -p casa-ms --test ms_library_coverage` passes.
- [x] `cargo clippy -p casa-ms --all-targets -- -D warnings` passes.
- [x] `cargo fmt --all -- --check` passes.

## Non-goals
- No new selector syntax
- No summary schema or text changes
- No `msexplore` or CLI option changes
- No behavior changes for valid selectors

## Stop conditions
Stop before implementation if the extraction starts to change user-visible selection semantics instead of only isolating parser/TAQL concerns and hardening failure-path coverage.

## Closeout

Implemented: moved listobs selector parsing and TAQL construction into `crates/casa-ms/src/listobs/selector.rs`, leaving summary/rendering logic in `listobs.rs` and preserving existing internal selector call points.
Changed from plan: none.
Deferred / follow-ups: none.
Verification run: `cargo test -p casa-ms --lib`, `cargo test -p casa-ms --test ms_library_coverage`, `cargo clippy -p casa-ms --all-targets -- -D warnings`, `cargo fmt --all -- --check`, `just quick`, `just verify`.
PR: #149.
Branch / worktree cleanup: deferred until merge via `wdad-pr-merge`.

### LINKED ISSUE #76: [Stabilization] Split casa-imaging trace and cube orchestration out of lib.rs
## Problem
`casa-imaging/src/lib.rs` remains an oversized library hotspot that mixes public trace APIs, cube orchestration, core helpers, and a very large in-file test module. That shape increases review cost and makes it harder to isolate orchestration regressions.

Hotspots:
- `crates/casa-imaging/src/lib.rs`
  - trace section
  - cube orchestration
  - in-file tests

## Scope
Extract the public trace surface and cube request-to-plane orchestration into narrower internal modules such as `trace.rs` and `cube.rs`, while keeping current public function signatures and behavior unchanged.

## Acceptance Criteria
- [x] The extracted trace and cube orchestration functions are no longer implemented directly in `lib.rs`.
- [x] Public APIs remain identical.
- [x] Test coverage for trace and cube behavior remains green after the split.
- [x] `cargo fmt --all -- --check` passes.
- [x] `cargo clippy -p casa-imaging --all-targets -- -D warnings` passes.
- [x] `cargo test -p casa-imaging --lib` passes.
- [x] `cargo test -p casa-imaging --lib trace_` passes.
- [x] `cargo test -p casa-imaging --lib run_cube` passes.

## Non-goals
- No new imaging modes
- No algorithm changes
- No public API redesign
- No `casars-imager` work
- No performance claims

## Stop conditions
Stop before implementation if the split would require changing public semantics rather than only moving internal ownership boundaries.

## Closeout

Implemented: moved public trace entrypoints and diagnostic conversion helpers into `crates/casa-imaging/src/trace.rs`; moved cube request-to-plane orchestration, clean cube orchestration, and cube-local helpers into `crates/casa-imaging/src/cube.rs`; preserved public crate-root exports.
Changed from plan: none.
Deferred / follow-ups: none.
Verification run: `cargo test -p casa-imaging --lib`, `cargo test -p casa-imaging --lib trace_`, `cargo test -p casa-imaging --lib run_cube`, `cargo clippy -p casa-imaging --all-targets -- -D warnings`, `cargo fmt --all -- --check`, `just quick`, `just verify`.
PR: #149.
Branch / worktree cleanup: deferred until merge via `wdad-pr-merge`.

### LINKED ISSUE #133: Split expansion-critical library hotspots not covered by existing stabilization issues
## Outcome

The largest expansion-critical library modules that are not already covered by existing stabilization issues are split into reviewable internal modules without changing public APIs or behavior, so tutorial functionality can be added at clear ownership seams.

## Included issues

- This complements, rather than replaces, #73, #75, #76, and #78.

## Non-goals

- Do not split `casa-imaging` trace/cube orchestration here; that is #76.
- Do not extract `listobs` selectors here; that is #75.
- Do not decouple curated visibility plots here; that is #73.
- Do not redesign algorithms or public APIs.

## Current understanding

Additional non-test hotspots over 3k LOC include `casa-tables/src/storage/tiled_stman.rs`, `casa-images/src/image_expr.rs`, `casa-images/src/image_view.rs`, `casa-vla/src/importer.rs`, `casa-calibration/src/cli.rs`, `casa-calibration/src/execute.rs`, `casa-tables/src/taql/functions.rs`, `casa-tables/src/storage/mod.rs`, `casa-images/src/image.rs`, and `casa-ms/src/spectral_selection.rs`. Some may be healthy; the wave should split only the ones where ownership boundaries are concrete.

## Acceptance checks

- [x] Behavior: Audit the listed hotspots and choose at most two first-wave extractions with the clearest ownership boundaries.
- [x] Behavior: Move implementation into narrower internal modules while preserving public symbols, serialized payloads, persisted bytes, and CLI/task behavior.
- [x] Edge case or failure mode: Existing regression and parity tests still exercise the moved code; no behavior is deleted to make the split easier.
- [x] Integration or contract: Public exports remain stable unless an explicit public API issue is approved.
- [x] Performance / UX / GUI expectation, if relevant: The split does not add extra allocation, full-row reads, or repeated parsing on tutorial data paths.

## Architecture / test impact

- Main area: internal module boundaries in expansion-critical libraries.
- Public API or persisted format change: no.
- Provider contract change: no.
- New dependency: no.
- ADR or docs needed: no unless a split reveals a true crate-boundary change.
- Verification to run: affected crate tests, targeted parity tests for moved behavior, `cargo clippy` for touched crates, `just quick`, `just verify` before review.

## Stop conditions

- Stop and ask if the split needs public API changes, persisted-format changes, or dependency-direction changes.
- Stop and split if more than two hotspot extractions are needed.

## Closeout

Implemented: audited the listed hotspots and selected one extra first-wave extraction: `casa-images` image-view plane rendering/statistics helpers moved into `crates/casa-images/src/image_view/plane_render.rs`. Dedicated child issues handled the `casa-ms` plot/listobs and `casa-imaging` splits.
Changed from plan: only one extra hotspot extraction was needed; the remaining candidates were deferred because their boundaries are either already owned by child issues or require later correctness-/interop-specific review.
Deferred / follow-ups: broader `spectral_selection`, tiled storage, image expression, VLA importer, calibration CLI, and TAQL function splits remain future focused work, not part of this wave.
Verification run: `cargo test -p casa-images --lib image_view`, `cargo clippy -p casa-images --all-targets -- -D warnings`, `just quick`, `just verify`.
PR: #149.
Branch / worktree cleanup: deferred until merge via `wdad-pr-merge`.

### LINKED ISSUE #135: Audit allocation and copy patterns on tutorial-scale library paths
## Outcome

Expansion-critical library paths have a concrete allocation/copy profile, and the first avoidable clone/collect/full-materialization patterns are fixed or split into performance waves before tutorial-scale data makes them expensive.

## Included issues

- This complements #94; #94 owns table/MS memory-bounded access architecture, while this issue audits broader library-level copies and collections on tutorial paths.

## Non-goals

- Do not micro-optimize cold code.
- Do not replace clear code with unsafe or overly clever code.
- Do not change algorithms or concurrency models without explicit approval.

## Current understanding

A quick non-test grep found over 2,000 clone/collect candidates, concentrated in `casa-tables`, `casa-images`, `casa-ms`, `casars`, `casars-imager`, and `casa-calibration`. The useful work is to identify which of these sit on tutorial-scale hot paths such as MS selection, plotting, calibration apply/solve, imaging orchestration, image analysis, and simulation generation.

## Acceptance checks

- [x] Behavior: Profile or statically classify allocation/copy hotspots on the first ALMA/VLA/Simulation tutorial paths.
- [x] Behavior: Fix at least one clearly avoidable clone/collect/materialization pattern in a covered hot path, or produce a dedicated performance issue when the fix is larger.
- [x] Edge case or failure mode: Changes preserve ownership correctness and do not introduce borrowed-lifetime contortions into public APIs unless explicitly approved.
- [x] Integration or contract: Any performance-sensitive API changes are reviewed as public API changes.
- [x] Performance / UX / GUI expectation, if relevant: Record before/after measurements for touched paths using existing or new reproducible scripts.

## Architecture / test impact

- Main area: performance hygiene across expansion-critical libraries.
- Public API or persisted format change: possible; stop and ask first.
- Provider contract change: no by default.
- New dependency: no by default.
- ADR or docs needed: no unless runtime/concurrency/performance algorithm changes.
- Verification to run: targeted perf scripts where available, affected crate tests, `just quick`, `just verify` before review.

## Stop conditions

- Stop and ask before changing runtime model, concurrency, major algorithms, or public API ownership shapes.
- Stop and split if a hotspot belongs to #94's table-access architecture rather than local allocation cleanup.

## Closeout

Implemented: classified tutorial-scale hotspot candidates and removed one avoidable clone pattern in the calibration apply path by consuming prefetched `DATA`, `FLAG`, and optional `WEIGHT` array-cell vectors with `into_iter()` instead of cloning each owned row payload.
Changed from plan: no dedicated performance follow-up was needed for the local cleanup; broader table-access architecture remains #94 and was not widened here.
Deferred / follow-ups: no new follow-up from this cleanup.
Verification run: `cargo test -p casa-calibration --lib`, `cargo clippy -p casa-calibration --all-targets -- -D warnings`, `just quick`, `just verify`.
PR: #149.
Branch / worktree cleanup: deferred until merge via `wdad-pr-merge`.

### LINKED ISSUE #139: Wave 2: Local seam stabilization
## Outcome

Clean the immediate seams needed by the first tutorial verticals: module hotspots, allocation/copy patterns, visibility plots, selectors, and imaging orchestration.

## Included issues

- [x] #133 Split expansion-critical library hotspots not covered by existing stabilization issues
- [x] #135 Audit allocation and copy patterns on tutorial-scale library paths
- [x] #73 [Stabilization] Decouple curated visibility plots from the generic msexplore scatter engine
- [x] #75 [Stabilization] Extract listobs selector parsing and TAQL building from summary/rendering
- [x] #76 [Stabilization] Split casa-imaging trace and cube orchestration out of lib.rs

## Implementation order

#133 -> #135 -> #73 -> #75 -> #76

## Non-goals

- Do not implement unrelated issues just because they share an area label.
- Do not change public APIs, persisted formats, provider-contract bundles, dependencies, or major algorithms without the stop-and-ask process in `AGENTS.md`.
- Do not move child issues to implementation before their own acceptance checks are still accurate.

## Current understanding

This is an umbrella issue. Child issues carry the implementation detail and acceptance criteria. This issue exists to keep the CASA tutorial expansion program pull queue coherent.

## Acceptance checks

- [x] Behavior: Child issues are implemented in the listed order unless closeout records a deliberate change.
- [x] Edge case or failure mode: If a child issue expands beyond its review boundary, split it before continuing the wave.
- [x] Integration or contract: Each child issue records its own API, provider-contract, persisted-format, dependency, and verification impact.
- [x] Performance / UX / GUI expectation, if relevant: Any performance-sensitive child issue records CASA/casacore comparison evidence or shapes the follow-up.

## Architecture / test impact

- Main area: wave coordination.
- Public API or persisted format change: child issues decide.
- Provider contract change: child issues decide.
- New dependency: child issues decide.
- ADR or docs needed: child issues decide.
- Verification to run: child issues define targeted gates; umbrella closes only after child closeout evidence is recorded.

## Stop conditions

- Stop and ask if wave execution would require reordering across numbered wave umbrellas.
- Stop and split if unrelated work is being pulled in to finish the umbrella.

## Closeout

Implemented: all listed Wave 2 child issues are implemented in PR #149. The wave extracted `casa-ms` curated visibility plot lowering, `listobs` selector/TAQL construction, `casa-imaging` trace/cube orchestration, and `casa-images` plane-rendering helpers; it also removed avoidable calibration apply-path array clones.
Changed from plan: #133 selected one extra hotspot extraction instead of two; broader candidates were deferred because their boundaries belong to later correctness-/interop-specific work. No public APIs, persisted formats, provider contracts, dependencies, runtime model, concurrency model, or major algorithms changed.
Deferred / follow-ups: broader `spectral_selection`, tiled storage, image expression, VLA importer, calibration CLI, and TAQL function splits remain future focused work, not part of this wave.
Verification run: `cargo test -p casa-ms --lib`, `cargo test -p casa-ms --test ms_library_coverage`, `cargo clippy -p casa-ms --all-targets -- -D warnings`, `cargo test -p casa-imaging --lib`, `cargo test -p casa-imaging --lib trace_`, `cargo test -p casa-imaging --lib run_cube`, `cargo clippy -p casa-imaging --all-targets -- -D warnings`, `cargo test -p casa-images --lib image_view`, `cargo clippy -p casa-images --all-targets -- -D warnings`, `cargo test -p casa-calibration --lib`, `cargo clippy -p casa-calibration --all-targets -- -D warnings`, `cargo fmt --all -- --check`, `git diff --check`, `just quick`, `just verify`.
PR: #149
Branch / worktree cleanup: deferred until merge via `wdad-pr-merge`.

### PR DESCRIPTION
**Pull request by bglenden**
Wave issue: #139

Closes #139
Closes #133
Closes #135
Closes #73
Closes #75
Closes #76

## Summary

- Extracted `casa-ms` curated raw-visibility plot lowering into an internal `plot_visibility` seam and added fixture-backed payload-shape coverage for amplitude/phase/UV-distance plots.
- Extracted `listobs` selector parsing and TAQL construction into `listobs/selector.rs`, with direct adversarial selector tests.
- Split `casa-imaging` trace APIs and cube orchestration from `lib.rs` into focused `trace` and `cube` modules while preserving crate-root exports.
- Split `casa-images` plane rendering/statistics helpers into `image_view/plane_render.rs` and removed avoidable calibration apply-path array clones by moving prefetched row inputs.
- Added `docs/tutorial-parity/wave-2-local-seam-audit.md` to record #133/#135 audit decisions and deferred hotspots.

## Verification

- `cargo test -p casa-ms --lib`
- `cargo test -p casa-ms --test ms_library_coverage`
- `cargo clippy -p casa-ms --all-targets -- -D warnings`
- `cargo test -p casa-imaging --lib`
- `cargo test -p casa-imaging --lib trace_`
- `cargo test -p casa-imaging --lib run_cube`
- `cargo clippy -p casa-imaging --all-targets -- -D warnings`
- `cargo test -p casa-images --lib image_view`
- `cargo clippy -p casa-images --all-targets -- -D warnings`
- `cargo test -p casa-calibration --lib`
- `cargo clippy -p casa-calibration --all-targets -- -D warnings`
- `cargo fmt --all -- --check`
- `git diff --check`
- `just quick`
- `just verify` (captured log completed through Rust workspace, doctests, and Python package tests: `19 passed in 13.05s`)

## Notes

No public APIs, persisted formats, provider contracts, dependencies, runtime model, concurrency model, or major algorithms were changed.

### CODE REVIEW

File: crates/casa-ms/src/plot_visibility.rs [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+                .get_mjd_seconds(row)
+                .map_err(|error| error.to_string())?,
+            VisibilityDomain::UvDistance => {
+                let [u, v, _w] = uvw.get(row).map_err(|error| error.to_string())?;
+                (u * u + v * v).sqrt()
+            }
+        };
+
+        let mut row_contributed = false;
+        for (corr_index, corr_label) in selected_correlations {
```
Comment (bglenden) [inline crates/casa-ms/src/plot_visibility.rs:153]: The new curated visibility seam changes the default payload shape for multi-correlation MS rows. The old generic msexplore path intentionally collapsed selected_correlations to the first correlation unless correlation was part of an axis/color/iteration choice; with the default AmplitudeVsTime, PhaseVsTime, and AmplitudeVsUvDistance specs that meant one point per row/bin. This loop now emits one point per selected correlation even when color_by is still field/SPW, so a dual-pol or four-pol dataset gets duplicate timestamps/UV distances in the same series and different point counts/rendered exports. That violates #73's acceptance that payload shape/grouping/rendered output remain unchanged. Either preserve the old first-correlation behavior unless color_by=correlation, or record and test an explicit behavior change and update the closeout.

Comment (bglenden) [reply crates/casa-ms/src/plot_visibility.rs:153]: Retracting this blocker after re-checking the old path: `MsAxis::Amplitude` and `MsAxis::Phase` both satisfy `uses_correlation_slots()`, so the previous generic lowering kept per-correlation samples for these curated plots. The new seam is not changing the multi-correlation shape in the way I described here.
